### ISSUE TITLE
doxygen -u always renames file in current directory to .bak

### ISSUE THREAD

**Issue by sternmull** [doxygen/doxygen#10035]
Updating a configuration of a file outside the current directory with the `-u`-option will rename the file with the same name in the current directory to ".bak". This can be reproduced with:

```
doxygen -g
mkdir foo
doxygen -g foo/Doxyfile
doxygen -u foo/Doxyfile
```

It should rename "foo/Doxyfile" to "foo/Doxyfile.bak" before doing the update. But it renames "Doxyfile" in the current directory before doing the update. It still correctly updates foo/Doxyfile.

**Comment by albert-github** [comment]
And after running `doxygen -u foo/Doxyfile` a second time the original `Doxyfile` is gone but also the `Doxyfile.bak`.

**Comment by albert-github** [comment]
Code (#10036) has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by sternmull** [comment]
That was quick! Thanks.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
